### ISSUE TITLE
base64 support

### ISSUE THREAD

**Issue by mib32** [jaimeiniesta/metainspector#166]
If you'll try to open specific page that has base64 image in it:

``` ruby
require 'metainspector'
i = MetaInspector.new('http://www.amazon.com/gp/product/B0107H5FJ6/ref=s9_simh_gw_g107_i3_r?pf_rd_m=ATVPDKIKX0DER&pf_rd_s=desktop-1&pf_rd_r=1DSA92Q4PADKNTZBWAZY&pf_rd_t=36701&pf_rd_p=2079475242')
i.images.best
```

You will get an error:

```
Errno::ENAMETOOLONG: File name too long @ rb_sysopen - data:image/jpeg;base64,/9j/4AAQS...
```

This is because MetaInspector is not taking base64 images into consideration. Do you have any options or plans to support that? Do you think that it is needed feature?

**Comment by jaimeiniesta** [comment]
Thanks for the report @mib32 but I couldn't reproduce this, it works on my machine and also on the demo server:

https://metainspectordemo.herokuapp.com/scrape?url=http%3A%2F%2Fwww.amazon.com%2Fgp%2Fproduct%2FB0107H5FJ6%2Fref%3Ds9_simh_gw_g107_i3_r%3Fpf_rd_m%3DATVPDKIKX0DER%26pf_rd_s%3Ddesktop-1%26pf_rd_r%3D1DSA92Q4PADKNTZBWAZY%26pf_rd_t%3D36701%26pf_rd_p%3D2079475242

I get http://g-ecx.images-amazon.com/images/G/01/gno/sprites/global-sprite_bluebeacon-32-v1._CB327533540_.png as the `images.best` URL.

Closing this, please reopen if we can reproduce it with another URL (and in that case, yes, a PR will be welcome).

Thanks!

**Comment by martijndekuijper** [comment]
Hey @jaimeiniesta, this is definitely still an issue. You can reproduce it:

```
require 'metainspector'
i = MetaInspector.new('http://www.newyorker.com/news/news-desk/the-desegregation-and-resegregation-of-charlottes-schools')
i.images.with_size
```

Any way to work around this? If I can filter out these svg images, I'm good.

Thanks.

**Comment by jaimeiniesta** [comment]
Thanks for reopening and providing more test URLs, I'll take a look as soon as I can.

**Comment by jaimeiniesta** [comment]
I've reproduced the bug in my machine, in MetaInspector 5.3.1.

It fails when trying to process an embedded image with FastImage, like this one:

```html
<img itemprop="image" src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/7QBeUGhvdG9zaG9wIDMuMAA4QklNBAQAAAAAACUcAVoAAxslRxwCAAACAAIcAnQAETIwMTYgR2V0dHkgSW1hZ2VzADhCSU0EJQAAAAAAEFuoUCT7g3BzKu9erY+fGAH/4QA2RXhpZgAASUkqAAgAAAABAJiCAgASAAAAGgAAAAAAAAAyMDE2IEdldHR5IEltYWdlcwAAAP/hBQNodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuNi1jMDE0IDc5LjE1Njc5NywgMjAxNC8wOC8yMC0wOTo1MzowMiAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wUmlnaHRzPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvcmlnaHRzLyIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczpkYz0iaHR0cDovL3B1cmwub3JnL2RjL2VsZW1lbnRzLzEuMS8iIHhtbG5zOnBob3Rvc2hvcD0iaHR0cDovL25zLmFkb2JlLmNvbS9waG90b3Nob3AvMS4wLyIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bXBSaWdodHM6TWFya2VkPSJGYWxzZSIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDo3ODRBNUIyRjdFMjkxMUU2ODFDQ0UzNTBCRUJGNzRGMyIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDo3ODRBNUIyRTdFMjkxMUU2ODFDQ0UzNTBCRUJGNzRGMyIgcGhvdG9zaG9wOkF1dGhvcnNQb3NpdGlvbj0iU3RyaW5nZXIiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENDIDIwMTQgTWFjaW50b3NoIj4gPHhtcE1NOkRlcml2ZWRGcm9tIHN0UmVmOmluc3RhbmNlSUQ9IjJFNzU4QjVBQkVDNTIwNUFCMjgzQjU2RDdDNUUzOTFFIiBzdFJlZjpkb2N1bWVudElEPSIyRTc1OEI1QUJFQzUyMDVBQjI4M0I1NkQ3QzVFMzkxRSIvPiA8ZGM6cmlnaHRzPiA8cmRmOkFsdD4gPHJkZjpsaSB4bWw6bGFuZz0ieC1kZWZhdWx0Ij4yMDE2IEdldHR5IEltYWdlczwvcmRmOmxpPiA8L3JkZjpBbHQ+IDwvZGM6cmlnaHRzPiA8ZGM6Y3JlYXRvcj4gPHJkZjpTZXE+IDxyZGY6bGk+U2VhbiBSYXlmb3JkPC9yZGY6bGk+IDwvcmRmOlNlcT4gPC9kYzpjcmVhdG9yPiA8ZGM6dGl0bGU+IDxyZGY6QWx0PiA8cmRmOmxpIHhtbDpsYW5nPSJ4LWRlZmF1bHQiPjkyNDg0MTM2PC9yZGY6bGk+IDwvcmRmOkFsdD4gPC9kYzp0aXRsZT4gPC9yZGY6RGVzY3JpcHRpb24+IDwvcmRmOlJERj4gPC94OnhtcG1ldGE+IDw/eHBhY2tldCBlbmQ9InIiPz7/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCABzAK0DAREAAhEBAxEB/8QAHAAAAgMBAQEBAAAAAAAAAAAABAYDBQcCAQgA/8QAPxAAAgEDAwIEAwYEAgkFAAAAAQIDBAURABIhBjETQVFhByJxFBUjgZGxMkKCoSTwCBYzUnLB0eHxQ2KiwtL/xAAaAQADAQEBAQAAAAAAAAAAAAACAwQFAQAG/8QALhEAAgIBAwMDAwIHAQAAAAAAAAECAxEEEiExQVEFEyIyYXGBoRQjQpGx0fDB/9oADAMBAAIRAxEAPwDAenau3x1iTXGOpkjjT5Y4CI8Htktg8fkdMU8ci5VuXxzgfFrrDXU8k9PRzSiFQzLBWnfjzOHTkj6D66BatqSU44Dfpy2N1zy/wWvWXT1FXdF09+tFdJXmjqBSVXiLiSBWQFFcAkcHPIOPmx5Y1qxab2mPtlBZYmW6DLAasqjkiuswaR0TaoqiqjEwGzzPtq76I5MfLtsURjr/AIu9C2TxqGkttVdJYRtM0e2ONm8wpOSfrjHprBt103P4vg+oo9MrUPkuSvpr9a+sr/S2qgtMtvnrNwhaacMC4z8pAUY7d+ffGqqPUe0+UR6v0V43V8MX+p7BU2qtkpq6neCZO6uMfmPUe+tNOFsd0HlGKvcolssWGKVXTAE6ROBfXbkq54O+p5RK4zApYdJcclEZAskWlNDkwZ00DQyMiFl76U1gamQsuhDTI2XQsMjZdCEcka4dOSNCzpyRrh05xrx4/EYOvHiWGdoZAy5wONvqPTXDwZRVjwsTCxWQnCkcHQyjljIza6G3Wu7WTp74b3uxtF41+vSxTSSRn8ONFZSqkZ4xzg+eefLVGitds9uMJZ/Ul9QrjXXuTy3gVrNTGWUAYC/zMeAo9TrWnrKdM0pvl9lyzDr0F+sTda4XVvhDjDWUtFZ61Kqq+zRyxPEJxnCkqcH15PH56d6vOdekbrXLwiD0SuFmuSsfCy/7GEVSxRz1CxSNIFfCMFIyPfOvnINtLJ9nJJN4NE6E6+mtdxt1ZVU0de1DKssUbP4XI4785J+gH112uEasyj3CsundFVyxhH1b1Fdumusfhgb5VInhyQM0G7HixTcjYPfcMY7HGrNE7YXKNff/AAZXqUaXRKVvbp5yfMdfDgnX0c4nytMylqI++pZxNCuRXypzqeSKoyA5U9tJkiiMgSRdKaGpgzroGh0WQuNKGJkTDQsNMiI0LDRGRoTp74blC4Rig7tjgfnoW1nAWH1I8ZPvrx4aKawxWnZJfImlqmG5bejbXC47yH+X/hxn1x5+bjFcs9HdN4iuBfr5op6l38FYsknah40OWFhFyXslcoMoanmKuWO3auTyoG0Hjy5GpsWR6clP8ufXgIk6WV5DNZq+GdFZQm8gZJfaBwe/Y444OhWoa4msHXpk+YMrPtlXbaiaHKCcNtldgr5wewPIx9NW1XOKzWRWUqTxZyOlbfo6KghllQCPA2Qx/LubHn3/AF1fTTXo4fxEvlN92Zuovt10/wCGh8a12QmX68VV3dZJSqxqNqxITtHv376kv1VmolmZZptFVpYtV9/7kFlmqkqkWiiaeRzgQhdxb2A1PLC5ZZDPYY+nbXQ3OtqJIahqe6KpaChKFQzgZzv7Ad+ONV6amN/wzz2IdbqXpUrMfHKz9v0GjoqK5xtOKld7VLnxd5Yujg4AyeCTz/f1Gq/T6LoW738YxznwZ/q2r09tHtLEpyax5QFe+pKaG4mnpYmq4lJDSocBjjkLkeXr/wCdNu9V5xWuPuJ03omI5tlz4Rq/wwsHQ/UHTErdQTzU1wqX/Cd5NngLnAAIO1jnk5H5aTLWW2YlBflFS9PqqTU2/sxV+J3w0ufRs32gH7ZZ5GxFVxjtnsHHkf7Hy9NPqvjcvD8E1unlQ/K8mbSprrR2LA5V50mSHxYLIulsdFg7jSpIaiFhoA0yJhoWMRGdCEPfSd2S60idOyIsL1EDUsfJKyknco9n3YIPbOO2pL6HKSsj1RXRfGMXXLowPpHp2WL4g2alrNqQLWIzvIMDYrBmBB7HAPB/bTabVZ06+BV1Mq/we3iVRca6ZSoEpZiXOWCk57nnU7e6TZRjasCdJTtJITB+Ivf5QePbVMYyl0RJKUU+Wd0ME9XULDTRtJI3YAZ0uclBZkNhGU3iIyL939MjxJ1Suu2Plj/9OE+p9TqT56nhcR/yV/y9Ny+ZfsikjMt5uLz1E8cckjZdnBA9zgDVaSrjhEjbtllvqF9S1iVs6ugUeFlCqDCghiM48sgKce+qrrnalkkpoVLljuynYYHGeRqdFDQXa6prfXU1VTkeLBIsihxkEg559tcktyaYUXtaaNOt/WHTl/ZE6itgop9oVKmFmYKQAAQ+fEXgdiWGTzjUvt2VPNUilzruW22J31fdILIlXSLXGvjeGOSmkJG51fcMbkBVsYHzcZzrZp9Utt07rtXPQ+ft9Fpq1MbqeEucfczWgrmpasVLqSWBGT2JPft5ajccrBqKTi8n0/8AAW42W8zRUV3pqbxXpWSMM/yndjgAgEEhiO50VSlWntZ7USVm3cuDa7VaIGsNT05dQtVSxKaYK/O+Ej5M+4XAz6qTpk5PcrI8EdceHVPn/R8V9W2sWbqO6WwPvFHVSQB/94KxGf7a1VLdFS8mTt2ycfAvyrpckPiCyDSmOQNIulSHRZAw0sYiFx30LGRIzoWEcqSrBlJBByCPLXDpqlq6kpb/AFlFWPIKe+KM1AkAK1EoOd4H/u4yPXPkRrO1NUoy31o0NNbFx2TYu9QdPTVclObbFNNJMHlnUHIiQSMoJ44Hy55/64vrrilBSeHIgttk5WNL4xKWvjp6eKnSkEijDFi55Jz/ANMa0pQjWkombXOVjbmQUl7eitv2agiEMzj8Wccs3Pl6awpUKc90+fsb0dQ4Q2wWH5KrIZiXJye50/oI69Rx+GEtPBfHFSiSxEZAdcjIBIOPXUurTdfBVosK3kMv9mpj17LKgWWhndagx54y5+ZTj3yfoRpVNk/4f7rgfdVD+J8p8/3JvjVYqey3+3SW+jSloaikUqIxhWcEhuPXBXXvT7nZBqTy0z3qVEarE4LCaEFHP+Rq7Bnpk0ZQAdwfUa4wlgIoTElbAKvP2Zv9pjuU9R766unALXKyNFk6RiqrYlyvFwS3WUsWRjh5Z8eSICefXyGk2alp7K1mX7Ior0qa9y14j+7Gzpj7os3WT1dyaYWukRwkcL5lzjanCkYPn3wMflooyk3vXAFkIpbHyfSnw/vMsdzmtNyqZa59sc0Nwfad6OXVVcrxu3I5BHGCBwe9Ltc1yQeyoSyj5U6xqvtnVF5qc58esmkz9XJ1spYikYmcybF2UaCXQdEFkGlMcgWQaVIdEgcaUMRA3noWMRERoQzlhrh1HsKhpkDEgFhkgcjXDrPob4zqnSdDRdOWyL8ARrJPWSL+LNuzgHngYXHvznuc30QTxZLqv2Mu695dMej5/JiVcizuviOq47H1021bnyDVLYuEUcdOxQuHA2+usY28EgikMJkMZKBtpPv/AJOhys4CSeMhdml+y1ELwtiV5BtJz8uD/f8A7aPYrFt7sBWOp7+yLStuyw1VORMJG3uZ8IQQc8Ht5duPIaOyEY1KqPYXXZKVzul3xgvev+sqXqbpuht4ikasomDioyNrDGCAO/p+ms3T6WVNrnnhmtqtZHUVKGOV3EBARtDjaSMjI76tZnr7l30903cL9JsoIPkQZedztijHqzHSLdRCr6n+hRTpp3fSv17FncbJSyz0do6aqJrzXqzfaJUULDk4ACE+Q5yTxrlds8Odq2rt5O21Ryq6nuffwVNmYrdaanbhXkALKc4XPzf2B1RJ4i5Ingk5KLNA6woqOm+7aq27YZTTOxcS7jIQi/MT6lmJxn0Gk0Sk1z5KL4JdPBcfC3q25rV0dmlqJA1TUR+CSxVYmLYLDHY4ZvbntnBF7ipLL6mXJyj06Hfxa6Cq+j69JyTLQVLEJJkEhu+D9cE/rrUjZGyOYmN7cq3tkZ3UxSRbfERk3KGXcMZB7Ee2hl0GxAZNKY5A8mlsdFgz6UMRzDTzVMhSmhklcAttjUscDucDXMZDzjqQSIyMVcFWBwQRyNAw0RHnQhE0cc1NKkpjIMbBsEdufMa5lZwdxlG0/H28y3K+UJracRVH2VG3rISHJ57dgOcjGcZwTnIGrFKEEjFit9jkY7O5D43YHtzpM5clkI8ACEhe/HbGso1EidKllpfBZCRuLZzwc+eNC485DUsLAfaIonWOVZYy8bYETrzyCSRx7frjTaM+8lgTqNvsyeSC8mI1UgUBSF5wOCcf99PvxveCfT59tZKtG+bGfbjU5QmX/TZts1xiW9RVE9KisVihYKWbHAJPlpFqnt/lvDKaXBy/mLKDup+qbnXf4CCCOgtUZxHR0wwn9R/mOgp0sK/k3l+WHdq7LPiliPhFTFcau2wOtLGYPFTHiAHOCBnB1RtUupNuceh50qzx3Q1KEh4I2cH+3/PXLuY4O08SyatYema3qaxm6Gpp0kjnaliiYrEJHaMlNvOMlgBjAz6+nakocHbpOQswUlf01fo5Z4Xpq+knRzFOpXsQ6nHmDt7g/TVNTTkmS3fQ19h2vFX1Z1/X9PQ3RClFcp/CpGjTEZIbYzE5JJXB7ngA4760Mxgm12MeKcml5Lv/AEnYKCjvthoaONUnpqAI4HlGDiMfltb9dIobcWUWxxLBiEmNGzyBZdLl0GxB3HOljENdlqZYrDNUQTmijBFI/gfKZVYEtvI75wOD+XbiXUt4WCzSqLk8lH1Na2tk9OPtEdRDNGJIpEPdfQjyI9NBp7vcTz1Q3U0qprHRkXSdRFRdS26qqIRMkMyyeGezsOVB9icapistIjm8RbC7asU1uudU0opaVE2yArvZyzfKF55Pc+XbU/tZll9ir3dsdqXUZPjLUR/6wwBJTLGlHACQylSTGG4wSc5JGDyMenA1JyxXFvwY9EPnNLuzOxV7ScxoPr31O7PJYq/DB1PGoyxEq9tCEXduSFKGJwJPHYkkkjbjPkO/lq7TwSW7uZuqsbbh2AuoFyY5CSTt2+wA/wDOualfJMLSv4tAcNSPsDUi00G53DeOwO8Y8gfTUTj8t2S5S+O3B0sYjYFTz++u5ydxgIjYrJE45QEEr6/loQl1yeVtY81EsWQEQ7EUZ555PJ74AGijwBJZ5LHpKnfw6qoC5OABxnPJJ4/TXpctI7WsJs1+0XC6wTUEFtqZKZTGXaJdvhvIQF37MFd3PBAyOPTWzLS1uuOVj4mHDVWRtnzn5YIOruqYfuiiNVRoLtS1UisJV3RTRNuD8epZh3PGBjWXscGa0Z7jTui+obVR/D/pWioKqge/moDpGJElNKWlPiMVBzwkjgDvz7Z0e+UotsmlTGE1gyD45101d8Tr1JUHlHSNR/uqEXA1XWsQSJZvdNszqTXWzqR5TUVTXSMlJDJMyruYIM4HqfQcjS5DY8AtVBLTSmKojaOQd1YYOlhoYOhoqWuqpKK9SinsZkSetmA+cBdwCpz/ABHeQP17AjQTxteRkcqSx1LmwW203Rb7TzxCCnQn7PuIZosthRu88ZH11j3TnCxSgbunrhZXKFhnLF6ap+RtskT8Mp7EHuDrWT7oxpLsxprY7KrSwNX+HBU0qzSbY9xjmxlUVRxjPfkYDEdxjTLNiWF1fURV7knmS4XC/wBnUdE3VVbYqSKqUBYlpHlZeYwuTuI7njtk9+PLTq3K2EV3XAu3bp5Skuj5EqaEJPKiMGVWKhjxnB76jksPBbF5SZFGcAg6ANMnU/IToQ0MLSI605izt8CMYPqEAb++daVbW1YMq1Pc8gF7b/DRnH82NL1HRDNN1ZVRAsMDuPbUbLkTFyVO7uByPXQhZPIJwcBgc+x11o8pBMqBWDgF1bDHP6H/AJaELHI3dE1qxUdZC5WORlZlIUYOQq4/LXov5YCxwx4sd0Wl6gtjeGCYRCH54Pzbjx9BrevntSj9kv8A0+f09e5ym11k3/4Z/wBRu7dPR1k6N+NIyjLZXJJb+4xrJnykzWhw2mO3wMtD091pbpcZEp6JZEczEjaZJOFX2JB7fXTK+FyIte7hA3xHjqOoOqeqL5Tqgp6ZoXnG7O0uFXA45+bP6afD6SWXEsMz6RtebDihv6esMd06HrAk4gmqqwJLM5AjihjUH5j3ALSA/wBA4Opb28ZRVp0nLDBI/h5fH6cmudLNSV1BSqz7oZcsyg/NsUgEgD5icY59cjSFqa4YjJ4bKJ6ayWZRWUioprQ1R0/SFKhUqbjcFp4IyDn5AAXz5DMuOfQ6rml7fJDGbVvHZFHe6iNK6r+73nFM0rCIyH5ymeM488Y1Mooucn3COn6WCsq5YJ0LymBzAviCNTIBkbiR2wDxxngZGmIRJ4KqpimhmYTRmNsnKkYKn0xoXw8MYuVlDd8NK2KkukslREslOphaozkYiEyB8EduD306qzam0TXVe44pmi2+ssnTdfdaC49PwVWyqcRsoUMoBIwSQcjgY+p1G7FGTT8lXtuSTiYGwZZCG/izzpjWDqDIYHmUrCju+MkKucD10tvHUao5XBaUystJThxhtnr7nWhV9KM61fJn6sgaopnVBllG8flrtizECqW2RSxEjjHftqGRoRJaxfDUKxGfbnQxeQ5LBDGMdyCNEAkWaoTTFiQwUdu/GlZ5wOxwd0TNURMAm48FQBwNclwzsOUSWm8VtJLKyVDcArlvmGCpXz9idUb5JdRChF5yFXK8x19lo7dsI2TB2x/CAqbePrk6LduWBbjiWTTLk0dF8GOnzb6mVHqLypryeOcMqAH0Crn8zoW8Swe2prcH9frTWXoauoaZvEmut2R1lAH4sMaHt6/MV1VCWcIksg9zZDe/g/FbbHaWkv8AFHfrhTiaOgmi2qzYz4YcE/N5DIwT6annqoxnsY+GmnKG9dig6K6rqbP0j1FYYaKM1EqSzTSSjJWMIFZAvrkAk+ikeemy6Co/Uhs+A6TXKhuhpvD+SKRcNwGZwMA+38WvmfUY7ZqTPqNHPNTiZx933G33ikp66nkhNkp6nxhjISUb2Ukjj5i0QB89fRqxWwjOHQ+a2e3ZKEura/7/ACIdc/8AiAq9k4z76Wilmj1E1v6klhrKKoht1VPH41xk+zbyhUAFl+v8R24ySc9hoYycFt+/7B2VqyW9eOn3E7q4H79q9s3jpIVnjfbtyjqGAxk4wCB3PbTr44nnzj/BLppbq0vGV/Z4BunqtaavAmbbTzo0EpPYKwxu/I4b8tB9hv3NKqWmqKKjr2jZ5JVME4HJWeLCvn6jY39Wp7IN8ofCSXBmV1kt0kkb21KtB/OKgqT+RGqZtPlE1akuJHNPWy08cq0zld67Xwcbh6aRKKfUojNroMMFguYtdLVxQiqp5l3BqZhIU8yGUEspGfMD2zq6pNRwZ904uTzwAykrFMCGVguMdiORorH8WcqXyTK63U0c1QBJIwXywOSdZ820uDQrSb5O+pKOeirBFUSQvIoGfCkDjkeo40NUlJZQd8XF4ZVKOx00QXViVpPEgTmSYFQDjGMZPc+g0qzjkfVysA8MktvrZVkGwqNraJpSQKk4S5AfEPhkebHJ00UWdijEhcttAXzPloo9TkuhoPUkdXR2OjtMc8jx/baKRFB4BeEsMc8HLt6d9Lae7cMbUYYLH4gU1TW19tijnRGo4FjD1Em1nlb5vDXjlhwPTkZxp6lGCJZKU3wHXYS9QdQW+ilq5pKilp4YDM5ZVWRVQNjI/h3Eny8/bUeppUW2jR0csxwz91x00ZuqL6bbOKmrrI5UZUlRRlQoY9+d0ivxxjHnnh9Mm4KD64IZxSbsXRMc/gZS26gp69aasoxJM/grTCceKypkiQoTnBDA51g+qRk/kumTd0LSik+uMkfxqrqeg6arFEY8a6SxozlMbjGwB57nAT/551pekTzoWvDMv1OrGvjLtj/v8ny3JzIx7cnjVgpjPDT001nh+7oqn7SYzHUOSCu4kFNoxxkow5PcjXcfLBxvEcsYOtaemfoTpuqipkWqFNEjzju4Cvx+WBp1jzBNk1CxZJIQAqAA9zqRyZqxrjFZNj+EHVk9ttdYsUUNRKzoJElA42ghWGfUcf0D10+qWESaiGWjF3i8NlViNxOeCCNKzk61gMhmRaSdXiR3ePapPBU5ByPfgj89DJZwFF4TCKGIvdIhF+GzYIPbBI8tOrTckkT2y21uT7DVd7bXvYZK2sqeUUKFmyxKnjhjznVttcva3yZj0ayt6lUwj17oU7Lbaq51IpaMDxTkrlgucAnz+msuyagss+hrg5vCBpVQI/2sy+IrY4xzoo4ccoGeVLDBEU+GWXkBgNdBL2zUtYsclckDNCgwxK/KM+WkznHO1lFcJJbkRdQTU1TAkscDQ1Awsu5shmGeR6DGNHWmvwDa0+nUq4kjNKzEkyFsY9BpmRXGCWjmSEhlLeIpBAxwfY65zk7xgZ6e7S3r7HQ18jl3qoUG0ZOEQon7gflo0+cHJcx+5rtRRWqZrRcJJd88tR9tNVNMyrSsCFyigeiBiTnJwAM6OUP6uwqMuq7vgzy1XOGhvZMEhmAlLCVlKlhn2PAIJ9dT2NtFtElBjD1NTWq222lqLS1bJXClNGjTgIQ8ruWIxyw2gkHP83POdHW3jK/Ai6CTUZdM5Er4SX6ksnX9LdLtO0VKu/e+0ucspUcD66i19MrKNkFyVaO2MLd03wO/x5uU1fZbdPHURGkllDRRI2ShIbfk59QP01z0mLhpX93/ALD9ValqI/Zf6MQA5GtAzxisV5uVPQ1VropMQVe13QD5g0bb1ZT5MMHn0J16Ta5R2KT4YzTtV3P4SR0ywsZLJcGhqOPmWOUM0bHzHzmQfpo8twx4EpKNu5dxBeMr8vpqZ9TSh8onqzSw/wCylZMjnacZ1480QMzMytgnjy0aI2EMFCEFSWYAKQcYPGuPqEugSssil9xzjhVYbgPpr209vZfUZuVba66aplla30SRgRuxwu9sKAO3kx7eWizNLa3wKca929RWfOOSr6dqnorvTVMQXxYpFdQwyCQfMaRdHdFplVMtsk0DdQ8VLBo9jBz3GM88j8tFU/iDd9QBGQsRyuGYg6YxSG7pqv8ABtlbTOiyRzBf4udhB741JbHMk/BbTPEWhc6jRBe6iOAYiDnYNuMAn08tUVP4rJPdxN4Bpo2ihijYEBxuHGPPv+g05rCQhPLZBBlWLY40KCLlqiOM0lXSBYamACQndklw2VP7D8tDFPnIcmuMGtdHTVFf0vRyzUSVZZ2ky8gygzs2gHsPk/fV1azDpkilPE8C51bbaeh6wqoqRcQblxjHfA3HA4HzbuNInDkOq145O/iRcEm6ZtklOmx4xTg7lwwYKx/tnH66ZKDrqTOK/wB2bh4yZar7ZmKjCk8DUr5KYtovuqJqitrp6ieQykxQPuxj5doXgeQydepqjVDbFcHbrXbPdJi8M7s++jAHX4f09GtTUVNxp1kAppTCWkwFJRlDD3B5GdclyuD0euH0ZYW/qBZ+rL3SVNQXpL1TJCzjjDqEdCP6l2/1adKWZPPcnUNsE0un+xWukSQXN6fnCgYLcf55zqSXk1NNL+lgFTC8D7XQj0yMa4uQ5tJg7Eqir8pXJI4Gf103sQ45D6GCesrqanoonmklYARR8l8c4wPPjtri5Z18LIV96MWzJbYiRn+GJR/9fbRbogbJeRmtlyhf4c9URClMJkqaLnyBHin9gfLXm0+hza11FFgYatSHwQR8w8tKlyh1b5OLwPHubJHK04J4cjBbnvjXKuInbn8iKrRYaoxbgVUbcjz01ikWnT07xiqRanwQ0fK8/icj5eP88antWccFVLazyVFU5mu0rccscj8tOgsIRY8thk0DtVwRAZZI923PltLY7/XRt5QtcFcOEPHIzxriCD1q/FiEropVFSPb5EA+f6a5CO07OTkNAoqmSkulZEJUaihiK7HK4ViefoAD+urXhLJDFuTX6lnHHNW0FbKqtJJbpEaRs8+FIBt9+/GfcaVuysjJR2ywD9fzpK1ZSKuFWKOePI5PP7bSdM1FmfiBp68fIzjb5+WcakLC/r3CfYRNuWGppFjcjvtDn/8AI0aYDWc/YkunTZprpPSU86yqkYkVs5DgqCMEex0E3seA4LfFM/WaEFVgdc743i3+SEg4OfYkaTZLHKKKop/FkDUrR9R0MVQjxMTEjhuCGAAOffTbJJw3R8CKotT2z65PbnCGtz1Lg+LJIGyQSX/iB/uufz1xJtZOt7XgKlpftFPTfaZdsqqQc8ZGcD9tKzhsZjKRbdcU0EFmgeGGONjJyUUDPHtrlDbzk9cksCdbyRVRkHByDnTG8co5FZNV+7aKWyR1MtJA9RJRJM8jRgszmIMWJ8yTzoLZNWJfg9CK9tv8iFWRqnTLlBt31jBsHg7VGOPbJ/XTH9QlFPGx8Rxns/GvTGQ6hUBIuzEHBEZYEeR269U+Dl3UrUG/eG5AUEaJgosLUf8Aaf8ACf20mY+ALSjfcl3c5LZ/TTV0Ey6llP8Ah1a7CVzTluD54I0Ue4DKmDmV8843ftrwRzEx8ELnjOcf5+uvHjTqqV06Vq9rY8SzwFvf8QD9idUTfxX6E9SxJ/qWnTKgzdaggEfYIzj6PFjSo9A7fqRX/EeniFxpXCDdshiz6qYiSP7a7azlC+P6mWyqFiUgYyx/fShxZXtifuxf5RTDAHH8zaPsgV1Y4Rgf65RxhVEa0sGFCgAfgjy/M69JJ5z9z0G0l+UH9NU0LSUaNGpVq2dGGO67O2s6xvD/AAjSqSyvyys6uhRbZFcAP8ZDJAiSEkkAiTPHY/wL39NV6eKdDz5I9TNrURx4Lew2miunw5nqa6DxZoYnKNuZcHxV8gQDo6P61/3YC76oP/urF/r2jp6f7k8GPZvttOzYJ5JjXJ1ySWT0W2j/2Q==" data-src-mobile="http://www.newyorker.com/wp-content/uploads/2016/09/Smith-Desegregation-Resegregation-Charlotte-Public-Schools-290x149-1475275755.jpg" data-src="http://www.newyorker.com/wp-content/uploads/2016/09/Smith-Desegregation-Resegregation-Charlotte-Public-Schools-690.jpg" class="feature-image-swap horizontal attachment-large" alt="The footage from Charlotte, North Carolina, following the killing, by police, of Keith Lamont Scott reflected a scene that has become all too familiar over the past several years." />
```

We need to investigate if there is a way in FastImage to support this, and if not, either avoid processing these kind of images, or better, try to use one of the alternative URLs provided in `data-src`.

**Comment by jschwindt** [comment]
I've just created a PR for skipping any non HTTP image url, but at the same time I'm asking Fastimage if they are interested in adding support for Data URI scheme: https://github.com/sdsykes/fastimage/issues/74

**Comment by jaimeiniesta** [comment]
Now that the PR for @jschwindt in FastImage has been accepted (great work!), I've checked the example for @mib32 and it works fine, it returns [this image](https://images-na.ssl-images-amazon.com/images/G/01/aplusautomation/vendorimages/0e2ee17d-476f-43ab-92d3-053d72f92933.jpg._CB526267614__SR970,300_.jpg).

The example by @martijndekuijper also works now, returning:

```ruby
[["http://www.newyorker.com/wp-content/uploads/2016/09/Smith-Desegregation-Resegregation-Charlotte-Public-Schools-690.jpg", 690, 461], ["http://condenast.112.2o7.net/b/ss/conde-newyorker/1/H.15.1--NS/0", 1, 1]]
```

**Comment by jaimeiniesta** [comment]
Just released MetaInspector 5.4.1, with base64 support. Thanks all!
